services/provider/server/server.go

```diff
@@ -907,15 +907,44 @@ func (s *OCSProviderServer) GetBlockPoolsInfo(ctx context.Context, req *pb.Block
 }
 
 func (s *OCSProviderServer) isSystemInMaintenanceMode(ctx context.Context) (bool, error) {
-	// found - false, not found - true
-	cephRBDMirrors := &rookCephv1.CephRBDMirror{}
-	cephRBDMirrors.Name = util.CephRBDMirrorName
-	cephRBDMirrors.Namespace = s.namespace
-	err := s.client.Get(ctx, client.ObjectKeyFromObject(cephRBDMirrors), cephRBDMirrors)
-	if client.IgnoreNotFound(err) != nil {
+	storageConsumers := &ocsv1alpha1.StorageConsumerList{}
+	err := s.client.List(ctx, storageConsumers, &client.ListOptions{Namespace: s.namespace})
+	if err != nil {
+		return false, err
+	}
+
+	// Step 1: Check if mirroring is enabled for any consumer
+	atLeastOneConsumerMirrorEnabled := false
+	for _, consumer := range storageConsumers.Items {
+		enabled, err := s.isConsumerMirrorEnabled(ctx, &consumer)
+		if err != nil {
+			return false, err
+		}
+		if enabled {
+			atLeastOneConsumerMirrorEnabled = true
+			break
+		}
+	}
+	if !atLeastOneConsumerMirrorEnabled {
+		return false, nil
+	}
```

Comment:
We could check if the data field in storage-client-mapping configMap is empty or not.

Reply:
Isn't that what the existing isConsumerMirrorEnabled func here does https://github.com/red-hat-storage/ocs-operator/blob/3ea6c860f7885b1e678b8242dc4e8c4036956481/services/provider/server/server.go#L921?

Reply:
isConsumerMirrorEnabled will load the configMap again and again from etdc, and since we are not using a cached client for the worst case scenario, it will hit the API server as many times we have the storageConsumer.

Reply:
Thanks. Updated now to avoid fetching the configmap again n again.